lib/spack/spack/cmd/flake8.py

```diff
@@ -127,7 +127,7 @@ def changed_files(args):
 
         for f in files:
             # Ignore non-Python files
-            if not f.endswith('.py'):
+            if not (f.endswith('.py') or f.endswith('spack')):
```

Comment:
Just make this `f == 'bin/spack'.  All the names from `git ls-files` are root-relative, and your current code is going to catch `var/spack/repos/builtin/packages/openfoam-com/common/README-spack` as well as the spack script.